### PR TITLE
Add VueJS documentation extension

### PR DESCRIPTION
**Pull request by Dpblandin**
## Description

Add a new extension for searching the VueJS documentation.
Most of the code is borrowed (thanks to him) from [indykoning's](https://github.com/indykoning) [Laravel docs extension](https://github.com/raycast/extensions/tree/main/extensions/laravel-docs) as it uses the same search engine (Agolia) under the hood
Can search either V2 or V3 documentation version via preferences.

## Checklist

- [X] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [X] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [X] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [X] I checked that files in the `assets` folder are used by the extension itself

### CODE REVIEW

File: extensions/vuejs-documentation/package.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  "author": "DavidPascaud",
+  "categories": [
+    "Documentation"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Search Documentation",
+      "subtitle": "VueJS Docs",
```
Comment (pernielsentikaer) [inline extensions/vuejs-documentation/package.json:16]:
```suggestion
      "subtitle": "Vue.js",
```

File: extensions/vuejs-documentation/package.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "vuejs-documentation",
+  "title": "VueJS documentation",
```
Comment (pernielsentikaer) [inline extensions/vuejs-documentation/package.json:4]:
```suggestion
  "title": "Searching Vue.js Documentation",
```

File: extensions/vuejs-documentation/package.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  "categories": [
+    "Documentation"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Search Documentation",
+      "subtitle": "VueJS Docs",
+      "description": "Search VueJS documentation from RayCast",
```
Comment (pernielsentikaer) [inline extensions/vuejs-documentation/package.json:17]:
```suggestion
      "description": "Search Vue.js documentation from Raycast",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @Dpblandin 👋 

Thanks for your contribution, did you notice that another one earlier was almost finish implementing this? (#675)

Maybe you can get some ideas, in that extension it also starts with a list, but I'm not sure if it makes sense, also, it's divided in multiple commands.

Above is inspiration, since it's also closed 😄  

Left a few NITs for you to address 💪 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by Dpblandin** [comment]
Hi @pernielsentikaer 👋 
Thank you for the review and suggestions.
As for #675 I did notice it right before making this PR. That extension does use 2 commands for searching either V2 or V3 docs, but I choose to use prefs for that because Vue V2 and V3 being very different, there not much of switching between them.

That being said i have added initial default lists for v2 / v3 👍

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @Dpblandin 👋 

Thanks for the update 🔥 

I checked your metadata image, and it seems like it's not in the right size, do you need help doing it? There is a guide here on how to make the [screenshot](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots)

In the newest version, `1.30.2`, of `@raycast/api` we introduced [search bar dropdown](https://developers.raycast.com/changelog#1.30.0-2022-03-09) - with this you should be able to do some magic with a fancy dropdown where it's possible to change version, it will look something like this:

uFP8JgzUnqCq9foN


Then you could move the version to Storage and let the extension read the version when opening, could that be an awesome idea? 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by Dpblandin** [comment]
Hello @pernielsentikaer 
Thank you for your review. I didn't know about the dropdown, very good addition! good job 🔥 

> Then you could move the version to Storage and let the extension read the version when opening, could that be an awesome idea?

Done 👍 
Also updated the screenshot, hope this one will be ok 🙏

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Wow, @Dpblandin 🔥 

It's turning out very nice!! Three small comments:

- Can we remove `navigationTitle` attribute, it should be the same as the command when only one view

- I'm looking from a phone right now, but `import { LocalStorage } from "@raycast/api";` doesn't seem to be used? Not sure where it's saved, screen is too small to have the overview 😄 

- If I select v2 and reopens the command, then I can see v3 loads first, before changing data to v2, do you think it's possible to use the loader to keep data showing up too early?

Already looking forward to testing it out again 💪 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by Dpblandin** [comment]
Hey @pernielsentikaer 
Thanks again for the review.
Removed navigation title and the unused import
> If I select v2 and reopens the command, then I can see v3 loads first, before changing data to v2, do you think it's possible to use the loader to keep data showing up too early?

I wasn't too sure about how to handle that, since the dropdown use the LocalStorage (i guess ?) to persist the vue version. Since it's async there is a short time during which vueVersion is undefined. So I made a "ghost/placeholder" list with a proper loader.
I'm open to a (surely) better solution.
Thank you!

**Review by pernielsentikaer — APPROVED** [review]
Hi @Dpblandin 👋 

Thanks a bunch for the update, it's turning out quite nice! 

I'm sending it to the final review before merge 🔥

**Pull request merged**